### ISSUE TITLE
Anchor tag-wrapped block level elements do not trigger link handling

### ISSUE THREAD

**Issue by cwhenderson20** [Selz/jquery.selz#6]
Wrapping an `<a>` tag around block-level elements is valid HTML5, but it does not correctly trigger the Selz link handling.

For example:
```html
<a href="SELZ-SHORT-LINK-HERE">
    <div class="row">
        Content Here
    </div>
</a>
```
will result in a console error message: `Woops. It looks like your link is to a product that can't be found!`. 

This appears to be happening because the click event that triggers `openOverlay()` is also being used to set the event context (i.e., `$trigger` = `$(event).target()`). 

It is naively assuming that the click event must have happened on the `<a>` tag itself rather than any content within it. Thus, when the `getItemData` method is called and the `href` attribute is extracted from the trigger, unless the trigger was _actually_ the `<a>` tag, the parsing will fail silently. 

Without modifying the javascript, the workaround is to add `href` attributes to all items inside of the `<a>` tag, but this is clearly ugly and non-semantic. Maybe the DOM should be traversed upwards until an enclosing `<a>` is found — then _it_ should be used as the "trigger"?

**Comment by SamPotts** [comment]
Hey @cwhenderson20 ,

Thanks for raising this. I'll look at fixing this very shortly :+1: 

Cheers
Sam

**Comment by cwhenderson20** [comment]
Hey @SamPotts,

No problem. Was a fun problem to figure out. Since you're using a selector filter on the `$(document.body)` click handler, I think the solution might actually be as easy as changing:
```javascript
$trigger = $(event.target);
```
 to be 
```javascript
$trigger = $(this);
```
on line 93 (in the `openOverlay` method). 

Hope this helps! :)

**Comment by SamPotts** [comment]
Yeah, I think you're right. I'll do some testing in a bit. Just working on something else right now. 

Cheers :+1:

**Comment by cwhenderson20** [comment]
No worries. Thanks!


On August 6, 2015 at 1:58:21 AM, Sam Potts (notifications@github.com) wrote:

Yeah, I think you're right. I'll do some testing in a bit. Just working on something else right now.

Cheers

**Comment by SamPotts** [comment]
This should be fixed in 1.0.7 :+1:
